perses/app/relative_point_mutation_setup.py

```diff
@@ -159,23 +159,31 @@ def __init__(self,
 
         # Load the ligand, if present
         molecules = []
-        if ligand_file:
-            if ligand_file.endswith('.sdf'): # small molecule
-                ligand_mol = createOEMolFromSDF(ligand_file, index=ligand_index)
-                ligand_mol = generate_unique_atom_names(ligand_mol)
+        if ligand_input:
+            if isinstance(ligand_input, str):
+                if ligand_input.endswith('.sdf'): # small molecule
+                        ligand_mol = createOEMolFromSDF(ligand_input, index=ligand_index)
+                        ligand_mol = generate_unique_atom_names(ligand_mol)
+                        molecules.append(Molecule.from_openeye(ligand_mol, allow_undefined_stereo=False))
+                        ligand_positions, ligand_topology = extractPositionsFromOEMol(ligand_mol),  forcefield_generators.generateTopologyFromOEMol(ligand_mol)
+                        ligand_md_topology = md.Topology.from_openmm(ligand_topology)
+                        ligand_n_atoms = ligand_md_topology.n_atoms
+
+                if ligand_input.endswith('pdb'): # protein
+                    ligand_pdbfile = open(ligand_input, 'r')
+                    ligand_pdb = app.PDBFile(ligand_pdbfile)
+                    ligand_pdbfile.close()
+                    ligand_positions, ligand_topology, ligand_md_topology = ligand_pdb.positions, ligand_pdb.topology, md.Topology.from_openmm(
+                        ligand_pdb.topology)
+                    ligand_n_atoms = ligand_md_topology.n_atoms
+
+            elif isinstance(ligand_input, list): # list of oemol object(s)
+                ligand_mol = ligand_input[0] # assume only one molecule in the list
```

Comment:
If we're assuming there is only molecule in the list, let's just allow the options for ligand_input to be a .pdb, .sdf or oemol object (passing a list thats always one elements feels unnecessary). You should be able to check if the `ligand_input` is an instance of `oechem.OEMol`

Reply:
don't know if reading a small molecule as a .pdb is safe, come to think of it

Reply:
the small molecule should not be passed as a pdb. if `ligand_input` is a protein, it should be passed as a pdb. this will be noted in function description (it's in one of my suggestions in this PR)